src/js/components/Calendar/Calendar.js

```diff
@@ -847,21 +846,24 @@ const Calendar = forwardRef(
         if (!children) {
           days.push(
             <CalendarDay
+              activeProp={active}
+              day={day}
               key={day.getTime()}
+              disabledProp={dayDisabled && !!dayDisabled}
               buttonProps={{
                 a11yTitle: day.toDateString(),
-                active: active && active.getTime() === day.getTime(),
-                disabled: dayDisabled && !!dayDisabled,
-                onClick: () => onClick(dateObject),
-                onMouseOver: () => setActive(dateObject),
-                onMouseOut: () => setActive(undefined),
+                onClick:
+                  dayDisabled && !!dayDisabled
+                    ? () => {}
+                    : () => onClick(dateObject),
```

Comment:
With the above suggestion this could be simplified:
```suggestion
                onClick: () => if (dayDisabled) onClick(dateObject),
```

Reply:
fixed in latest commit